lib/identity/partialClient.js

```diff
@@ -54,25 +55,34 @@ class PartialClient extends CryptoConsumer {
           expiry_seconds: 1,
         },
       }
-      const request = await anonAuth.tsv1Fetch(
+      const response = await anonAuth.tsv1Fetch(
         `${this.config.apiUrl}/v2/storage/notes/challenge?id_string=${noteName}`,
         {
           method: 'PATCH',
           'Content-Type': 'application/json',
           body: JSON.stringify(body),
         }
       )
-      await checkStatus(request)
-      return true
-    } catch (_) {
-      return false
+      await checkStatus(response)
+
+      ret = {success: true, message: ""}
+      return ret
+    } catch (e) {
+      switch (e.response.status) {
```

Comment:
The Identity Portal update password flow does not check for any error objects, so in order to make minimal changes we check the possible cases here.

Reply:
you should check that `e.response` exists before this switch and return false with no message